tests/api/test_api_siteinfo.py

```diff
@@ -0,0 +1,21 @@
+import json
+
+import pytest
+import requests
+
+
+@pytest.mark.parametrize(
+    "url,status_code,validator",
+    [
+        pytest.param(
+            "http://localhost:5000/get-location-values/45.551538/-73.744616?dataset_name=cmip5",
+            200,
+            json.loads,
+            id="get_location_values"
+        )
+    ]
+)
+def test_api_siteinfo(url: str, status_code: int, validator: callable):
+    response = requests.get(url)
+    assert response.status_code == status_code
+    validator(response.content)
```

Comment:
This part is to make sure that `response.content` is in the right format?

If the response is always the same for this url, an assertion against a dictionary could be a better approach and be even more informative (output is as expected vs `json.loads` didn't cause an error). 

That, or assert that certain keys are present in the `response.content`.

Also, since it is likely that other tests would use the same validator, I'd be tempted to use this approach instead, and exceptions could have a separate test.

```
    response = requests.get(url)
    content = json.loads(response.content)
    assert response.status_code == status_code
    assert content == get_location_values_content # expected dictionary, or else some other check
```

Reply:
@perronld Have a look, now only JSON validity is checked, wrapped up in a function so failure of this test will be easier to interpret than a `ValueError` in the test logs